### DISCUSSION TITLE
Whisper gives different results based on the platform

### DISCUSSION THREAD

**Discussion by Tharuja** [openai/whisper#1336] [Q&A] [answered]
I have tried the Whisper model with numpy==1.24.3 version in both Linux-AMD64 and MacOs M1-ARM systems.. Both give two different results as shown below (with different no_speech_prob values)... What is the reason for this? I used Python 3.8 version.

Results in MacOs M1-ARM

`{
        "text": " The number one biggest difference between the bacteria in your body and the cells making of your body are these tiny cellular components called organelles. You've actually learned a lot about organelles in other lessons without knowing it."
        "language": "en",
        "segments": [
            {
                "id": 0,
                "seek": 0,
                "start": 0.0,
                "end": 5.0,
                "text": " The number one biggest difference between the bacteria in your body and the cells making",
                "tokens": [440, 1230, 472, 3880, 2649, 1296, 264, 11763, 294, 428, 1772, 293, 264, 5438, 1455]
                "temperature": 0.0,
                "avg_logprob": -0.23756315901472763,
                "compression_ratio": 1.5868544600938967,
                "no_speech_prob": 0.018022961914539337, 
            },
            {
                "id": 1,
                "seek": 0,
                "start": 5.0,
                "end": 9.24,
                "text": " of your body are these tiny cellular components called organelles.",
                "tokens": [295, 428, 1772, 366, 613, 5870, 29267, 6677, 1219, 1798, 19787,13]                
                "temperature": 0.0,
                "avg_logprob": -0.23756315901472763,
                "compression_ratio": 1.5868544600938967,
                "no_speech_prob": 0.018022961914539337,
            },
            {
                "id": 2,
                "seek": 0,
                "start": 9.24,
                "end": 14.08,
                "text": " You've actually learned a lot about organelles in other lessons without knowing it."
                "tokens": [509, 600, 767, 3264, 257, 688, 466,1798, 19787, 294, 661, 8820, 1553, 5276, 309, 13]       
                "temperature": 0.0, 
                "avg_logprob": -0.23756315901472763,
                "compression_ratio": 1.5868544600938967,
                "no_speech_prob": 0.018022961914539337,    
            },
        ],
    }`

Results in Linux-AMD64 :- 

`{
    'text': " The number one biggest difference between the bacteria in your body and the cells making of your body are these tiny cellular components called organelles. You've actually learned a lot about organelles in other lessons without knowing it.", 
    'language': 'en'
    'segments': [
      {
        'id': 0,
        'seek': 0,
        'start': 0.0,
        'end': 5.0,
        'text': ' The number one biggest difference between the bacteria in your body and the cells making', 
        'tokens': [50364, 440, 1230, 472, 3880, 2649, 1296, 264, 11763, 294, 428, 1772, 293, 264, 5438, 1455, 50614], 
        'temperature': 0.0, 
        'avg_logprob': -0.25806166330973307, 
        'compression_ratio': 1.610878661087866,
        'no_speech_prob': 0.002833886770531535
      }, 
      {
        'id': 1, 
        'seek': 0, 
        'start': 5.0, 
        'end': 9.24, 
        'text': ' of your body are these tiny cellular components called organelles.', 
        'tokens': [50614, 295, 428, 1772, 366, 613, 5870, 29267, 6677, 1219, 1798, 19787, 13, 50826], 
        'temperature': 0.0, 
        'avg_logprob': -0.25806166330973307, 
        'compression_ratio': 1.610878661087866, 
        'no_speech_prob': 0.002833886770531535
      }, 
      {
        'id': 2, 
        'seek': 0, 
        'start': 9.24, 
        'end': 14.08, 
        'text': " You've actually learned a lot about organelles in other lessons without knowing it.", 
        'tokens': [50826, 509, 600, 767, 3264, 257, 688, 466, 1798, 19787, 294, 661, 8820, 1553, 5276, 309, 13, 51068], 
        'temperature': 0.0, 
        'avg_logprob': -0.25806166330973307, 
        'compression_ratio': 1.610878661087866, 
        'no_speech_prob': 0.002833886770531535
       }, 
      ], 
}`

**Comment by phineas-pta** [comment, marked as answer]
it's the default behaviour see #81
